src/wilcoxon.jl

```diff
@@ -142,14 +142,14 @@ function StatsAPI.pvalue(x::ExactSignedRankTest; tail=:both)
         # Compute exact p-value using method from Rmath, which is fast but cannot account for ties
         if tail == :both
             if x.W <= n * (n + 1)/4
-                2 * psignrank(x.W, n, true)
+                2 * psignrank(typeof(n)(x.W), n, true)
```

Comment:
Since `n` is always an `Int` here, this just means
```suggestion
                2 * psignrank(Int(x.W), n, true)
```
it seems? But AFAICT that would error if `x.W` is non-integer (it's stored as a `Float64` it seems).

What's the current Rmath implementation actually doing for non-integer values of `W`? Probably we just have to round, either here or at the top of the function body of `psignrank`. Since it has to be done in multiple places it might be easier to add a `round(Int, ...)`/`floor(Int, ...)/...` statement to `psignrank`.